app/app/src/main/kotlin/com/hedvig/app/feature/profile/ui/payment/PaymentDestination.kt

```diff
@@ -0,0 +1,321 @@
+import java.util.*
+import kotlinx.datetime.LocalDate
+
+@Composable
+fun PaymentDestination(navigateUp: () -> Unit) {
+
+}
+
+@Composable
+fun PaymentScreen(
+  uiState: PaymentUiState,
+  locale: Locale,
+  navigateUp: () -> Unit,
+) {
+  Box(
+    modifier = Modifier.fillMaxSize(),
+  ) {
+    HedvigScaffold(
+      topAppBarText = stringResource(R.string.PROFILE_PAYMENT_TITLE),
+      navigateUp = navigateUp,
+      modifier = Modifier.clearFocusOnTap(),
+    ) {
+      Column(modifier = Modifier.padding(horizontal = 16.dp)) {
+        Spacer(Modifier.height(16.dp))
+        NextPayment(uiState, locale)
+        Spacer(Modifier.height(4.dp))
+        Divider()
+        Spacer(Modifier.height(4.dp))
+        InsuranceCosts(uiState, locale)
+        Spacer(Modifier.height(4.dp))
+        TotalDiscount(uiState)
+        Spacer(Modifier.height(4.dp))
+        Divider()
+        Spacer(Modifier.height(16.dp))
+        AddDiscount(uiState)
+        Spacer(Modifier.height(32.dp))
+        PaymentDetails(uiState)
+        Spacer(Modifier.height(4.dp))
+        Divider()
+        PaymentHistory(onClick = {})
+        Spacer(Modifier.height(16.dp))
+        HedvigContainedButton(
+          text = stringResource(id = R.string.PROFILE_PAYMENT_CHANGE_BANK_ACCOUNT),
+          onClick = {},
+        )
+      }
+    }
+  }
+}
+
+@Composable
+private fun NextPayment(uiState: PaymentUiState, locale: Locale) {
+  HedvigCard(modifier = Modifier.fillMaxWidth()) {
+    Text(
+      text = uiState.nextChargeAmount.format(locale),
+      textAlign = TextAlign.Center,
+      style = MaterialTheme.typography.displayMedium,
+      modifier = Modifier.padding(vertical = 6.dp),
+    )
+  }
+  Spacer(Modifier.height(12.dp))
+  Row(
+    modifier = Modifier
+      .fillMaxWidth()
+      .padding(vertical = 12.dp),
+    horizontalArrangement = Arrangement.SpaceBetween,
+  ) {
+    Text(stringResource(R.string.PAYMENTS_NEXT_PAYMENT_SECTION_TITLE))
+    Text(
+      text = uiState.nextChargeDate.toString(),
+      color = MaterialTheme.colorScheme.onSurfaceVariant,
+    )
+  }
+}
+
+@Composable
+private fun InsuranceCosts(
+  uiState: PaymentUiState,
+  locale: Locale,
+) {
+  uiState.insuranceCosts.forEach { insuranceCost ->
+    Row(
+      modifier = Modifier
+        .fillMaxWidth()
+        .padding(vertical = 12.dp),
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically,
+    ) {
+      Row(
+        verticalAlignment = Alignment.CenterVertically,
+      ) {
+        Image(
+          painter = painterResource(id = com.hedvig.android.core.ui.R.drawable.ic_pillow),
+          contentDescription = null,
+          modifier = Modifier.size(32.dp),
+        )
+        Spacer(Modifier.width(8.dp))
+        Text(insuranceCost.displayName)
+      }
+      Text(
+        text = insuranceCost.cost.format(locale),
+        color = MaterialTheme.colorScheme.onSurfaceVariant,
+      )
+    }
+    Divider()
```

Comment:
Is the divider at the end of the list here on purpose?
Maybe the `InsuranceCosts` component by itself can only render the dividers between its items, and if the entire layout does require a divider there for some reason it can be part of the column of items inside `PaymentScreen` instead?
Basically what I mean is so that `InsuranceCosts` itself does not make any assumptions about where it's placed, it just works no matter where it's placed. Right now I *think* that it makes the assumption that the item below it also wants a divider between that and this InsuranceCosts component. So if that items leaves from the `PaymentScreen`, then this composable would need to be edited, which makes this a bit non-obvious.

Another way to think about it, is consider if you wanted to reuse this composable from another screen completely. You'd again want it to just render itself, and not make any assumptions about what's around it.

Does my thinking make sense? Maybe the design literally here asks for the divider at the end, even if it's the last item, but in that case the design is most likely wrong.

Reply:
NextPayment right above for example does this well I think. It does not have the divider at all, but the screen composable itself knows that it wants a divider between it and the InsuranceCosts, so the screen puts it instead.